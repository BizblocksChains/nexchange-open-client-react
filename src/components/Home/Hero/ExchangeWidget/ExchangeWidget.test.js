import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ExchangeWidget from './ExchangeWidget.js';

describe('ExchangeWidget', () => {
  const initialState = {
    wallet: { address: '', valid: false, show: false },
    selectedCoin: { deposit: 'OMG', receive: 'ETH', prev: { deposit: 'OMG', receive: 'ETH' }, lastSelected: 'deposit' },
    price: { pair: 'ETHOMG', deposit: 101, receive: 1.76250023, lastEdited: 'deposit' },
    error: { show: false, type: 'INVALID_AMOUNT' },
  };
  const mockStore = configureStore();
  let store, wrapShallow;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapShallow = shallow(<ExchangeWidget store={store} />)
      .dive()
      .dive();
  });

  it('renders correctly', () => {
    expect(wrapShallow).toMatchSnapshot();
  });

  it('contains CoinInput, WalletAddress', () => {
    expect(wrapShallow.find('Connect(CoinInput)').length).toBe(2);
    expect(wrapShallow.find('Connect(WalletAddress)').length).toBe(1);
  });

  it('submit button changes on wallet shown', () => {
    expect(wrapShallow.find('button').text()).toEqual('exchangewidget.2');
  });
});
