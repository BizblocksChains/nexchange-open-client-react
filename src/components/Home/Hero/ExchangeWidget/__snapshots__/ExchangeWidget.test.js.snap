// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExchangeWidget renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
>
        <div
                className="row"
        >
                <div
                        className="col-xs-12"
                >
                        <div
                                className="widget"
                        >
                                <Connect(CoinInput)
                                        onSubmit={undefined}
                                        type="deposit"
                                />
                                <Connect(CoinInput)
                                        onSubmit={undefined}
                                        type="receive"
                                />
                                <Connect(WalletAddress)
                                        inputRef={[Function]}
                                        onSubmit={[Function]}
                                />
                                <div
                                        className="submit"
                                >
                                        <p
                                                className="info"
                                        >
                                                order.feeinfo
                                        </p>
                                        <button
                                                className="btn disabled btn btn-block btn-primary proceed "
                                                onClick={[Function]}
                                        >
                                                exchangewidget.2
                                        </button>
                                </div>
                        </div>
                </div>
        </div>
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
>
          <div
                    className="col-xs-12"
          >
                    <div
                              className="widget"
                    >
                              <Connect(CoinInput)
                                        onSubmit={undefined}
                                        type="deposit"
                              />
                              <Connect(CoinInput)
                                        onSubmit={undefined}
                                        type="receive"
                              />
                              <Connect(WalletAddress)
                                        inputRef={[Function]}
                                        onSubmit={[Function]}
                              />
                              <div
                                        className="submit"
                              >
                                        <p
                                                  className="info"
                                        >
                                                  order.feeinfo
                                        </p>
                                        <button
                                                  className="btn disabled btn btn-block btn-primary proceed "
                                                  onClick={[Function]}
                                        >
                                                  exchangewidget.2
                                        </button>
                              </div>
                    </div>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-xs-12"
>
            <div
                        className="widget"
            >
                        <Connect(CoinInput)
                                    onSubmit={undefined}
                                    type="deposit"
                        />
                        <Connect(CoinInput)
                                    onSubmit={undefined}
                                    type="receive"
                        />
                        <Connect(WalletAddress)
                                    inputRef={[Function]}
                                    onSubmit={[Function]}
                        />
                        <div
                                    className="submit"
                        >
                                    <p
                                                className="info"
                                    >
                                                order.feeinfo
                                    </p>
                                    <button
                                                className="btn disabled btn btn-block btn-primary proceed "
                                                onClick={[Function]}
                                    >
                                                exchangewidget.2
                                    </button>
                        </div>
            </div>
</div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="widget"
>
              <Connect(CoinInput)
                            onSubmit={undefined}
                            type="deposit"
              />
              <Connect(CoinInput)
                            onSubmit={undefined}
                            type="receive"
              />
              <Connect(WalletAddress)
                            inputRef={[Function]}
                            onSubmit={[Function]}
              />
              <div
                            className="submit"
              >
                            <p
                                          className="info"
                            >
                                          order.feeinfo
                            </p>
                            <button
                                          className="btn disabled btn btn-block btn-primary proceed "
                                          onClick={[Function]}
                            >
                                          exchangewidget.2
                            </button>
              </div>
</div>,
            "className": "col-xs-12",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Connect(CoinInput)
                  onSubmit={undefined}
                  type="deposit"
/>,
                <Connect(CoinInput)
                  onSubmit={undefined}
                  type="receive"
/>,
                <Connect(WalletAddress)
                  inputRef={[Function]}
                  onSubmit={[Function]}
/>,
                <div
                  className="submit"
>
                  <p
                                    className="info"
                  >
                                    order.feeinfo
                  </p>
                  <button
                                    className="btn disabled btn btn-block btn-primary proceed "
                                    onClick={[Function]}
                  >
                                    exchangewidget.2
                  </button>
</div>,
              ],
              "className": "widget",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onSubmit": undefined,
                  "type": "deposit",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "onSubmit": undefined,
                  "type": "receive",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "inputRef": [Function],
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="info"
>
                      order.feeinfo
</p>,
                    <button
                      className="btn disabled btn btn-block btn-primary proceed "
                      onClick={[Function]}
>
                      exchangewidget.2
</button>,
                  ],
                  "className": "submit",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "order.feeinfo",
                      "className": "info",
                    },
                    "ref": null,
                    "rendered": "order.feeinfo",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "exchangewidget.2",
                        null,
                      ],
                      "className": "btn disabled btn btn-block btn-primary proceed ",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": Array [
                      "exchangewidget.2",
                      null,
                    ],
                    "type": "button",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
>
          <div
                    className="row"
          >
                    <div
                              className="col-xs-12"
                    >
                              <div
                                        className="widget"
                              >
                                        <Connect(CoinInput)
                                                  onSubmit={undefined}
                                                  type="deposit"
                                        />
                                        <Connect(CoinInput)
                                                  onSubmit={undefined}
                                                  type="receive"
                                        />
                                        <Connect(WalletAddress)
                                                  inputRef={[Function]}
                                                  onSubmit={[Function]}
                                        />
                                        <div
                                                  className="submit"
                                        >
                                                  <p
                                                            className="info"
                                                  >
                                                            order.feeinfo
                                                  </p>
                                                  <button
                                                            className="btn disabled btn btn-block btn-primary proceed "
                                                            onClick={[Function]}
                                                  >
                                                            exchangewidget.2
                                                  </button>
                                        </div>
                              </div>
                    </div>
          </div>
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="row"
>
            <div
                        className="col-xs-12"
            >
                        <div
                                    className="widget"
                        >
                                    <Connect(CoinInput)
                                                onSubmit={undefined}
                                                type="deposit"
                                    />
                                    <Connect(CoinInput)
                                                onSubmit={undefined}
                                                type="receive"
                                    />
                                    <Connect(WalletAddress)
                                                inputRef={[Function]}
                                                onSubmit={[Function]}
                                    />
                                    <div
                                                className="submit"
                                    >
                                                <p
                                                            className="info"
                                                >
                                                            order.feeinfo
                                                </p>
                                                <button
                                                            className="btn disabled btn btn-block btn-primary proceed "
                                                            onClick={[Function]}
                                                >
                                                            exchangewidget.2
                                                </button>
                                    </div>
                        </div>
            </div>
</div>,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-xs-12"
>
              <div
                            className="widget"
              >
                            <Connect(CoinInput)
                                          onSubmit={undefined}
                                          type="deposit"
                            />
                            <Connect(CoinInput)
                                          onSubmit={undefined}
                                          type="receive"
                            />
                            <Connect(WalletAddress)
                                          inputRef={[Function]}
                                          onSubmit={[Function]}
                            />
                            <div
                                          className="submit"
                            >
                                          <p
                                                        className="info"
                                          >
                                                        order.feeinfo
                                          </p>
                                          <button
                                                        className="btn disabled btn btn-block btn-primary proceed "
                                                        onClick={[Function]}
                                          >
                                                        exchangewidget.2
                                          </button>
                            </div>
              </div>
</div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="widget"
>
                <Connect(CoinInput)
                                onSubmit={undefined}
                                type="deposit"
                />
                <Connect(CoinInput)
                                onSubmit={undefined}
                                type="receive"
                />
                <Connect(WalletAddress)
                                inputRef={[Function]}
                                onSubmit={[Function]}
                />
                <div
                                className="submit"
                >
                                <p
                                                className="info"
                                >
                                                order.feeinfo
                                </p>
                                <button
                                                className="btn disabled btn btn-block btn-primary proceed "
                                                onClick={[Function]}
                                >
                                                exchangewidget.2
                                </button>
                </div>
</div>,
              "className": "col-xs-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Connect(CoinInput)
                    onSubmit={undefined}
                    type="deposit"
/>,
                  <Connect(CoinInput)
                    onSubmit={undefined}
                    type="receive"
/>,
                  <Connect(WalletAddress)
                    inputRef={[Function]}
                    onSubmit={[Function]}
/>,
                  <div
                    className="submit"
>
                    <p
                                        className="info"
                    >
                                        order.feeinfo
                    </p>
                    <button
                                        className="btn disabled btn btn-block btn-primary proceed "
                                        onClick={[Function]}
                    >
                                        exchangewidget.2
                    </button>
</div>,
                ],
                "className": "widget",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onSubmit": undefined,
                    "type": "deposit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "onSubmit": undefined,
                    "type": "receive",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "inputRef": [Function],
                    "onSubmit": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="info"
>
                        order.feeinfo
</p>,
                      <button
                        className="btn disabled btn btn-block btn-primary proceed "
                        onClick={[Function]}
>
                        exchangewidget.2
</button>,
                    ],
                    "className": "submit",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "order.feeinfo",
                        "className": "info",
                      },
                      "ref": null,
                      "rendered": "order.feeinfo",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "exchangewidget.2",
                          null,
                        ],
                        "className": "btn disabled btn btn-block btn-primary proceed ",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": Array [
                        "exchangewidget.2",
                        null,
                      ],
                      "type": "button",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
